src/qtools/rad.cpp

```diff
@@ -381,15 +388,24 @@ void CalcFaceExtents(Bsp* bsp, lightinfo_t* l)
 			float shift = j == 0 ? tex->shiftS : tex->shiftT;
 
 			val = v->x * axis.x + v->y * axis.y + v->z * axis.z + shift;
-			if (val < mins[j])
+			if (!initminmax)
 			{
 				mins[j] = val;
+				maxs[j] = val;
 			}
-			if (val > maxs[j])
+			else
 			{
-				maxs[j] = val;
+				if (val < mins[j])
+				{
+					mins[j] = val;
+				}
+				if (val > maxs[j])
+				{
+					maxs[j] = val;
+				}
 			}
 		}
+		initminmax = true;

```

Comment:
would rather this `initminmax` stuff be removed entirely. The rad code is magical and I don't to risk breaking anything.

Reply:
Ok now changes reverted back!

Reply:
Thanks! If this project ever gets unit tests then changes like this will be less risky.